### PR TITLE
rename sint32 gFirstTimeSave to bool gFirstTimeSaving + bug fix

### PR DESCRIPTION
**Pull request by wolfreak99**
Because gFirstTimeSave sounds like it could mean either the first save has been executed or the user is saving for the first time, i have renamed it to gFirstTimeSaving for the sake of clarity*. 

In addition to that, I have moved a call to gFirstTimeSaving to the game_load_save(path) functionality. The call being in the callback function caused loading scenario files or sv4 files to be marked false, when only the sv6 files should be marked false. Setting gFirstTimeSaving should in general be something that should be initially set during the loading of the files anyway, and not reside in a dialog callback. I must have missed this back when i added the ability to load scenarios / s4 files from the in game load dialog.

*although i'm not sure how everyone feels about a present-tense terminology, i'm starting to feel weary about it myself.. it sounds like it could mean the user is currently saving for the first time, should i rename it to something like gIsNewGame ?

### CODE REVIEW

File: src/openrct2/game.c [hunk 1]
```diff
@@ -1083,10 +1083,12 @@ bool game_load_save(const utf8 *path)
     bool result = false;
     if (extension_type == FILE_EXTENSION_SV6) {
         result = game_load_sv6_path(path);
+        if (result)
```
Comment (Gymnasiast) [inline src/openrct2/game.c]: We left out this call deliberately here. When the user has loaded from an SV6, it's okay to overwrite that during the first quick save, while you can't do that with an SV4 file.

File: src/openrct2/game.c [hunk 2]
```diff
@@ -1357,8 +1359,8 @@ bool game_load_save_or_scenario(const utf8 * path)
 
 static void game_load_or_quit_no_save_prompt_callback(sint32 result, const utf8 * path)
 {
-    if (result == MODAL_RESULT_OK && game_load_save_or_scenario(path)) {
-        gFirstTimeSave = 0;
+    if (result == MODAL_RESULT_OK) {
+        game_load_save_or_scenario(path);
```
Comment (Gymnasiast) [inline src/openrct2/game.c]: What is the reason for this change?

File: src/openrct2/game.c (hunk 1 above)
Comment (wolfreak99) [inline src/openrct2/game.c]: Yeah, this also fixes that. the gFirstTimeSaving being true for sv4s will force the first save to show a "save as" dialog, while sv6 would mark it false saying it's okay to overwrite the loaded file

File: src/openrct2/game.c (hunk 2 above)
Comment (wolfreak99) [inline src/openrct2/game.c]: with the edits i did in the game_load_save(path), game_load_save_or_scenario now handles setting the gFirstTimeSaving on a per-file basis. https://github.com/wolfreak99/OpenRCT2/blob/0dcbacb09b52f96382159d325a63abcb265f9644/src/openrct2/game.c#L1084-L1091

The original method caused sc6 files to mark gFirstTimeSave to be true during game_load_save_or_scenario, but then reset them after exiting game_load_save_or_scenario.

Comment (Gymnasiast) [inline src/openrct2/game.c]: I see. game_load_save_or_scenario() calls either game_load_save() scenario_load_and_play_from_path(), both of which indeed set gFirstTimeSave.
